apps/ui/components/application/ApplicationEventList.tsx

```diff
@@ -51,112 +66,204 @@ const formatDurationSeconds = (seconds: number, t: TFunction): string => {
   )}`;
 };
 
+const getDuration = (begin: number, end: number, t: TFunction): string => {
+  const beginHours = formatDurationSeconds(begin, t);
+  const endHours = formatDurationSeconds(end, t);
+  return `${beginHours} - ${endHours}`;
+};
+
 const ageGroupToString = (ag: Maybe<AgeGroupNode> | undefined): string => {
   if (!ag) {
     return "";
   }
   return `${ag.minimum} - ${ag.maximum}`;
 };
 
+const getLabelProps = (
+  status: ApplicationSectionStatusChoice | undefined | null
+): { type: StatusLabelType; icon: JSX.Element } => {
+  switch (status) {
+    case ApplicationSectionStatusChoice.Handled:
+      return { type: "success", icon: <IconCheck /> };
+    case ApplicationSectionStatusChoice.Rejected:
+      return { type: "error", icon: <IconCross /> };
+    default:
+      return { type: "info", icon: <IconQuestionCircleFill /> };
+  }
+};
+
 // NOTE: used by Preview and View
 // No form context unlike the edit pages, use application query result
 type Node = NonNullable<ApplicationQuery["application"]>;
-export function ApplicationEventList({ application }: { application: Node }) {
+export function ApplicationEventList({
+  application,
+}: {
+  application: Node;
+}): JSX.Element {
   const { t } = useTranslation();
 
   const aes = filterNonNullable(application.applicationSections);
-  const reservationUnits = aes.map((evt) =>
-    evt?.reservationUnitOptions?.map((eru, index) => ({
-      pk: eru.reservationUnit?.pk ?? 0,
-      priority: index,
-      nameFi: eru.reservationUnit?.nameFi ?? undefined,
-      nameSv: eru.reservationUnit?.nameSv ?? undefined,
-      nameEn: eru.reservationUnit?.nameEn ?? undefined,
-    }))
-  );
 
+  const sections = aes.map((applicationEvent, i) => {
+    const primaryTimes =
+      applicationEvent.suitableTimeRanges
+        ?.filter(filterPrimary)
+        .map(convertApplicationSchedule) ?? [];
+    const secondaryTimes =
+      applicationEvent.suitableTimeRanges
+        ?.filter(filterSecondary)
+        .map(convertApplicationSchedule) ?? [];
+    const reservationUnits =
+      applicationEvent.reservationUnitOptions?.map((eru, index) => ({
+        pk: eru.reservationUnit?.pk ?? 0,
+        priority: index,
+        nameFi: eru.reservationUnit?.nameFi ?? undefined,
+        nameSv: eru.reservationUnit?.nameSv ?? undefined,
+        nameEn: eru.reservationUnit?.nameEn ?? undefined,
+      })) ?? [];
+    const { type, icon } = getLabelProps(applicationEvent.status);
+    return (
+      <ApplicationSection key={applicationEvent.pk}>
+        <ApplicationSectionHeader>
+          {applicationEvent.name}
+          {/* Show status label only for rejected and handled applicationEvents */}
+          {applicationEvent.status ===
+            ApplicationSectionStatusChoice.Rejected ||
+            (applicationEvent.status ===
+              ApplicationSectionStatusChoice.Handled && (
```

Comment:
use a variable instead of chaining booleans inside JSX (removes unnecessary code nesting)